pilot/pkg/model/service.go

```diff
@@ -417,6 +422,14 @@ type IstioEndpoint struct {
 	DiscoverabilityPolicy EndpointDiscoverabilityPolicy `json:"-"`
 }
 
+// GetLoadBalancingWeight returns the LbWeight for this endpoint, or DefaultLoadBalancingWeight if not set.
+func (ep *IstioEndpoint) GetLoadBalancingWeight() uint32 {
+	if ep.LbWeight > 0 {
+		return ep.LbWeight
+	}
+	return DefaultLoadBalancingWeight
```

Comment:
Have you seen a case where we have to fallback to this? If so that's an issue elsewhere and we might want a Debug log here.

Reply:
oh this was supposed to be reverted.